### PR TITLE
docs: simplify readme

### PR DESCRIPTION
**Pull request by theol0403**
The readme organization got a little too messy and complex over time. This removes items which are not necessary and improves formatting.

Additionally, the table of contents was not kept up to date. I use "Markdown All In One" to automatically update the readme.

### CODE REVIEW

**Review by xiyaowong — COMMENTED** [review]
A little flaw: https://github.com/theol0403/vscode-neovim/pull/28

**Review by xiyaowong — APPROVED** [review]
A little flaw: https://github.com/theol0403/vscode-neovim/pull/28

**Comment by ollien** [comment]
One small suggestion, which is what I would have left on #2148 

> Open the vscode-neovim extension details in VSCode, and check all available [settings](https://code.visualstudio.com/docs/getstarted/settings) and commands in the Features tab.

could be better written as 

> You can view all available settings and commands [settings](https://code.visualstudio.com/docs/getstarted/settings) and commands by opening the `vscode-neovim` extension details pane, and navigating to the features tab.

**Review by ollien — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -21,53 +21,67 @@ mode and editor commands, making the best use of both editors.
...
+\[Optional\] Set the Neovim path in the extension settings under
+"`vscode-neovim.neovimExecutablePaths.win32/linux/darwin`", respective to your system. For example,
+"`C:\Neovim\bin\nvim.exe`" or "`/usr/local/bin/nvim`".
 
-    -   Add to your `settings.json`:
-        ```json
-        "extensions.experimental.affinity": {
-            "asvetliakov.vscode-neovim": 1
-        },
-        ```
```
Comment (ya2s) [inline README.md:70]: Is this setting no longer necessary?

Comment (ya2s) [inline README.md:70]: @theol0403 
Is this setting no longer necessary?

Comment (theol0403) [reply README.md:70]: We add it automatically on setup

**Pull request merged**